connaisseur/validate.py

```diff
@@ -136,11 +137,11 @@ def process_chain_of_trust(
     # repository, but no signatures were created using the delegations. in this special
     # case, the releases JSON doesn't exist yet and the targets JSON must be used instead
     if req_delegations:
-        if not all(trust_data[target_role] for target_role in req_delegations):
+        if not all(trust_data.get(target_role) for target_role in req_delegations):
             tuf_roles = [
                 target_role
                 for target_role in req_delegations
-                if not trust_data[target_role]
+                if not trust_data.get(target_role)
```

Comment:
Can it happen that `target_role` is not in `trust_data`? If so, `get()` returns `None`, and the `if` conditions all evaluate to true. Is that intended?

If it can't happen, it will be safer to use `[]` subscripts, and if it really doesn't happen in practice, it will also never throw a `KeyError`. :)

Reply:
The way I re-wrote it calls `trust_data[delegation] = delegation_trust_data` only `if delegation_trust_data is not None:` (as part of setting `trust_data` entries only after validation [and if delegation data is None, we dont validate]), thus it can actually not exist and then correctly `get`s None
But maybe that's somewhat unexpected and I should set `None` inside the `trust_data` and stick to `[]`? Works in this context since validation is only skipped for `None` and that's a "safe" value in this context. Not sure :man_shrugging:

Reply:
ah, I see now (did not all of that in my working memory at the same time :-) ). I slightly lean towards unindenting `trust_data[delegation] = delegation_trust_data` by one level and sticking to `[]`, but it's probably not worth the effort.